redfish-exporter/redfish_utils.go

```diff
@@ -63,22 +63,30 @@ func getRedfishClient(server RedfishServer) (*gofish.APIClient, error) {
 	return c, nil
 }
 
-// Create a subscription
-func createSubscription(server RedfishServer, SubscriptionPayload SubscriptionPayload) (string, error) {
-
-	// Establish a connection to the server
+func getEventService(server RedfishServer) (*redfish.EventService, error) {
 	c, err := getRedfishClient(server)
 	if err != nil {
-		return "", fmt.Errorf("failed to connect to server %s: %v", server.IP, err)
+		return nil, fmt.Errorf("failed to connect to server %s: %v", server.IP, err)
 	}
 	defer c.Logout()
```

Comment:
Oh, I just realized this could introduce a bug. Since we log out immediately after the event service is returned (deferred), it might cause issues later when we try to interact with the event service. So [this](https://github.com/nod-ai/ADA/pull/14#issuecomment-2393658984) might not be a good idea after all.